### ISSUE TITLE
Assert.assertThat() had @deprecated, suggest use org.hamcrest.MatcherAssert.assertThat(),but dependency version is too old

### ISSUE THREAD

**Issue by ProYI** [junit-team/junit4#1665]
**Environment:**
JUnit 4.13
Java 1.8.0_201

when I use JUnit to check Exception, I find one problem:
```java
@Test
    public void testCheck() {
        try {
            checkNull(null);
        } catch (Exception e) {
            assertThat(e, is(instanceOf(NullPointerException.class)));
            assertThat(e, isA(NullPointerException.class)); 
        }
    }
```
because org.junit.Assert had be mark up @deprecated, so these I import org.hamcrest.MatcherAssert.assertThat.

but org.hamcrest.CoreMatchers.is also outdated, suggest use isA(Class<T> type) instead
![image](https://user-images.githubusercontent.com/26431420/86021852-2c3a5780-ba5c-11ea-823a-322dde1daae6.png)
this is result.
![image](https://user-images.githubusercontent.com/26431420/86021983-5429bb00-ba5c-11ea-83bf-4bc0bb76e82f.png)

I find JUnit 4.13 depend on hamcrest-core version is  1.3, this is hamcrest's maintainer [record](https://github.com/hamcrest/JavaHamcrest/issues/27#issuecomment-650830800).

**experiment**
1、change hamcrest-core version to 2.2
```xml
<dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>4.13</version>
            <scope>test</scope>
            <exclusions>
                <exclusion>
                    <groupId>org.hamcrest</groupId>
                    <artifactId>hamcrest-core</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>org.hamcrest</groupId>
            <artifactId>hamcrest-core</artifactId>
            <version>2.2</version>
        </dependency>
```
2、import org.hamcrest.Matchers.isA
this problem did not appear.

this problem had be solved at hamcrest-core 2.x,  It is recommended to upgrade the hamcrest-core dependent version

**Comment by stefanbirkner** [comment]
TLDR; JUnit will not change the version of Hamcrest that it depends upon.

Our plan was to remove the dependency to Hamcrest completely (see #1145). Unfortunately that is not easy. In the meantime JUnit Jupiter was released and we recommend developers to switch to JUnit Jupiter. Therefore the plan to remove Hamcrest is abandoned.
Changing the Hamcrest version in JUnit's dependency may cause problems and therefore it stays at 1.3. It is always possible to specify the version of Hamcrest in your pom.xml and override the version specified by JUnit.

**Comment by stefanbirkner** [comment]
FYI we recommend to use [`assertThrows`](https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertThrows(java.lang.Class,%20org.junit.function.ThrowingRunnable)) for testing exceptions:

    @Test
    public void testCheck() {
        assertThrows(
            NullPointerException.class,
            () -> checkNull(null)
        );
    }
